packages/react/src/components/Forms/Select/multi-simple.tsx

```diff
@@ -58,10 +76,21 @@ export const SelectMultiSimple = forwardRef<SelectHandle, SubProps>(
           case useSelect.stateChangeTypes.ToggleButtonKeyDownSpaceButton:
           case useSelect.stateChangeTypes.ItemClick:
             if (newSelectedItem) {
-              props.onSelectedItemsChange([
-                ...props.selectedItems,
-                newSelectedItem,
-              ]);
+              if (isSelected(newSelectedItem)) {
+                // Remove the item if it is already selected.
+                props.onSelectedItemsChange(
+                  props.selectedItems.filter(
+                    (selectedItem) =>
+                      optionToValue(selectedItem) !==
+                      optionToValue(newSelectedItem),
```

Comment:
Brainsto: having a utils function to compare equality of two options? Not sure whether it's a good idea.